src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useRef } from 'react';
+import cn from 'classnames';
+
+import { TodosContext } from '../../utils/TodosContext';
+import { State, Todo } from '../../type/type';
+
+interface Props {
+  todo: Todo;
+}
+
+const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { title, completed } = todo;
+  const { todos, setTodos } = React.useContext<State>(TodosContext);
+
+  const [inputValue, setInputValue] = React.useState(title);
+  const [isEditing, setIsEditing] = React.useState(false);
+
+  const inputField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputField.current) {
+      inputField.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleCheckbox = () => {
+    setTodos(
+      todos.map(prevTodo => {
+        return prevTodo.id === todo.id
+          ? { ...prevTodo, completed: !prevTodo.completed }
+          : prevTodo;
+      }),
+    );
+  };
+
+  const handleDeleteTodo = () => {
+    setTodos(todos.filter(prevTodo => prevTodo.id !== todo.id));
+  };
+
+  const handleAddNewTitle = () => {
+    if (inputValue === '') {
```

Comment:
```suggestion
    if (!inputValue) {
```
- You don't need to compare with an empty string, just check if exists https://developer.mozilla.org/en-US/docs/Glossary/Falsy
- The same about naming, fix it